org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java

```diff
@@ -314,43 +308,43 @@ protected Object getLastModified(IResource resource) throws CoreException {
 	protected PackageFragmentRootData initializeData(final IPackageFragmentRoot root) {
 		final PackageFragmentRootData data = new PackageFragmentRootData(computeModificationStamp(root));
 		data.addRoot(root);
-		try {
-			final SourceAttachmentPackageFragmentRootWalker<Void> walker = new SourceAttachmentPackageFragmentRootWalker<Void>() {
-				
-				@Override
-				protected URI getURI(IFile file,
-						org.eclipse.xtext.ui.resource.PackageFragmentRootWalker.TraversalState state) {
-					if (!uriValidator.isPossiblyManaged(file))
-						return null;
-					return super.getURI(file, state);
-				}
+		if (shouldHandle(root)) {
+			try {
+				final SourceAttachmentPackageFragmentRootWalker<Void> walker = new SourceAttachmentPackageFragmentRootWalker<Void>() {
 
-				@Override
-				protected URI getURI(IJarEntryResource jarEntry,
-						org.eclipse.xtext.ui.resource.PackageFragmentRootWalker.TraversalState state) {
-					if (!uriValidator.isPossiblyManaged(jarEntry))
-						return null;
-					final URI uri = locator.getURI(root, jarEntry, state);
```

Comment:
why is this no longer needed?

Reply:
I think the diff tricked you. See https://github.com/eclipse/xtext-eclipse/pull/1775/files?diff=split&w=1 with ignored whitespace changes.

Reply:
Or I misunderstood the question.